src/main/java/fi/helsinki/cs/tmc/core/commands/SendSpywareDiffs.java

```diff
@@ -30,22 +31,34 @@ public SendSpywareDiffs(TmcSettings settings, DiffSender sender, byte[] spywareD
     }
 
     private void assertHasRequiredData() throws TmcCoreException {
-        String username = settings.getUsername();
-        if (username == null || username.isEmpty()) {
-            throw new TmcCoreException("username must be set!");
+        testUsername();
+        testPassword();
+        testCourse();
+    }
+
+    private void testCourse() throws TmcCoreException {
+        Optional<Course> course = this.settings.getCurrentCourse();
+        if (course == null || !course.isPresent()) {
+            throw new TmcCoreException("No current course found from settings.");
         }
+    }
 
-        String password = settings.getPassword();
-        if (password == null || password.isEmpty()) {
+    private void testPassword() throws TmcCoreException {
+        if (isBadString(settings.getPassword())) {
             throw new TmcCoreException("password must be set!");
         }
+    }
 
-        Optional<Course> course = this.settings.getCurrentCourse();
-        if (course == null || !course.isPresent()) {
-            throw new TmcCoreException("No current course found from settings.");
+    private void testUsername() throws TmcCoreException {
+        if (isBadString(settings.getUsername())) {
+            throw new TmcCoreException("username must be set!");
         }
     }
 
+    private boolean isBadString(String toBeTested) {
```

Comment:
Also, "bad" is a bit unclear. +1 to replacing with statically imported `isNullOrEmpty(String string)`.